packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -166,10 +166,27 @@ export class Session extends EventEmitter {
 			// tslint:disable-next-line:no-console
 			log.error('network', `[ERROR] setAuthoritativeClient: client ${clientId} does not exist.`);
 		}
-		this._clientSet[clientId].setAuthoritative(true);
-		this.clients
-			.filter(client => client.id !== clientId && client.isJoined())
-			.forEach(client => client.setAuthoritative(false));
+		const oldAuthority = this.authoritativeClient;
+		const newAuthority = this._clientSet[clientId];
+
+		newAuthority.setAuthoritative(true);
+
+		// forward network stats from the authoritative peer connection to the app
+		const toApp = this.conn instanceof EventedConnection ? this.conn : null;
+		const toNewAuthority = newAuthority.conn instanceof EventedConnection ? newAuthority.conn : null;
+		const forwardIncoming = (bytes: number) => toApp.statsTracker.recordIncoming(bytes);
+		const forwardOutgoing = (bytes: number) => toApp.statsTracker.recordOutgoing(bytes);
+		toNewAuthority.statsTracker.on('incoming', forwardIncoming);
+		toNewAuthority.statsTracker.on('outgoing', forwardOutgoing);
+
+		// turn off old authority
+		const toOldAuthority = oldAuthority && oldAuthority.conn instanceof EventedConnection
+			? oldAuthority.conn : null;
+		if (oldAuthority) {
+			oldAuthority.setAuthoritative(false);

```

Comment:
You changed the behavior here. We used to send .setAuthoritative(false) to all other clients. You might be introducing a bug here (or exposing one). Safer to keep it the way it was.